include/tools/base64.h

```diff
@@ -1,134 +1,37 @@
- * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
- * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
- * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
- * OF THE POSSIBILITY OF SUCH DAMAGE.
- * ====================================================================
- *
- * This software consists of voluntary contributions made by many
- * individuals on behalf of the Apache Group and was originally based
- * on public domain software written at the National Center for
- * Supercomputing Applications, University of Illinois, Urbana-Champaign.
- * For more information on the Apache Group and the Apache HTTP server
- * project, please see <http://www.apache.org/>.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  *
  */
 
 #ifndef _BASE64_H_
 #define _BASE64_H_
-
-#ifndef __cplusplus
-#include <stdlib.h>
-#else /* __cplusplus */
-#include <climits>
-#include <cstdint>
-#include <cstdlib>
 #include <string>
-#endif /* __cplusplus */
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-int Base64encode_len(int len);
-int Base64encode(char* coded_dst, const char* plain_src, int len_plain_src);
-
-size_t Base64decode_len(const char* coded_src);
-int Base64decode(char* plain_dst, const char* coded_src);
 
-#ifdef __cplusplus
-}
-#endif
-
-#ifdef __cplusplus
-namespace cloudevents {
+namespace tools {
 class base64 {
- public:
-  static std::string encode(const std::string& t_str) {
-    size_t sz_size = Base64encode_len(t_str.length());
-
-    char* pc_dst = new char[sz_size];
-    if (nullptr == pc_dst) {
-      return "";
-    }
-
-    size_t sz_str_length = t_str.length();
-    if (sz_str_length > INT_MAX) {
-      return "";
-    }
-
-    int i_val = Base64encode(pc_dst, t_str.c_str(),
-                             (int)sz_str_length); /* TODO: safe convert */
-    if ((((size_t)i_val) != (sz_size - 4)) && (((size_t)i_val) != sz_size)) {
-      return "";
-    }
-
-    std::string ret_string(pc_dst);
-    delete[] pc_dst;
-    pc_dst = nullptr;
-    return ret_string;
-  }
-
-  static std::string decode(const std::string& t_str) {
-    size_t sz_size = Base64decode_len(t_str.c_str());
-
-    char* pc_str = (char*)new char[sz_size];
-    if (nullptr == pc_str) {
-      return "";
-    }
-
-    Base64decode(pc_str, t_str.c_str());
-    std::string t_ret_str(pc_str);
-    delete pc_str;
-    pc_str = nullptr;
-    return t_ret_str;
-  }
-
- private:
-  base64() = default;
+    public:
+        static std::string base64encode(uint8_t const* data, const size_t &len);
+        static std::string base64decode(std::string_view data, const size_t &len);
```

Comment:
why did you change it to string_view??

Reply:
The string_view is similar to const char * , I thought the string_view is faster compared to const char *, it is taking fixed length and no need to search null character for every time

Reply:
please revert back to std::string , we are not using string_view anywhere currently